packages/pirania/files/usr/lib/lua/voucher/vouchera.lua

```diff
@@ -40,21 +41,21 @@ function voucher_init(obj)
     end
     voucher.mac = obj.mac
 
-    if obj.expiration_date == nil and obj.duration_m == nil then
-        return nil
-    elseif obj.expiration_date ~= nil then
-        voucher.expiration_date = obj.expiration_date
-    elseif obj.duration_m ~= nil then
-        voucher.duration_m = obj.duration_m
-    end
+    voucher.duration_m = obj.duration_m -- use nil to create a permanent voucher
+
+    voucher.expiration_date = obj.expiration_date
```

Comment:
Wouldn't be less error prone if expiration_date is a derived property of creation_date + duration_m (if duration_m is not nil).

Reply:
This way also preserve the information of when was the voucher activated, even if has no duration_m.
In the other way: activation would be expiration_date - duration (when available)

Reply:
That seems a good idea!

Reply:
Done